File: apps/api/src/main/java/com/lifttrack/api/domain/models/WorkoutSession.java
package com.lifttrack.api.domain.models;

import java.time.Instant;
import java.time.LocalDate;
import java.util.UUID;

public record WorkoutSession(
        UUID uuid,
        UUID userUuid,
        UUID routineDayUuid,
        LocalDate sessionDate,
        Instant startedAt,
        Instant finishedAt,
        String notes,
        Instant createdAt) {

    public WorkoutSession create(UUID userUuid, UUID routineDayUuid, LocalDate sessionDate, Instant startedAt,
            Instant finishedAt, String notes) {
        if (userUuid == null) {
            throw new IllegalArgumentException("User UUID cannot be null");
        }
        if (sessionDate == null) {
            throw new IllegalArgumentException("Session date cannot be null");
        }

        return new WorkoutSession(null, userUuid, routineDayUuid, sessionDate, startedAt, finishedAt, notes, null);
    }
}

File: apps/web/src/domain/models/workout-session.ts
export interface WorkoutSession {
  readonly uuid: string
  readonly userUuid: string
  readonly routineDayUuid: string | null
  readonly sessionDate: string
  readonly startedAt: string
  readonly finishedAt: string | null
  readonly notes: string | null
  readonly createdAt: string
}

export interface CreateWorkoutSession {
  readonly routineDayUuid?: string
  readonly sessionDate: string
  readonly startedAt: string
  readonly finishedAt?: string
  readonly notes?: string
}
